src/app/main.py

```diff
@@ -23,12 +23,24 @@ def new_dataset(cloud_event):
     bucket_name = cloud_event.data["bucket"]
     filename = cloud_event.data["name"]
 
-    dataset = dataset_storage.get_dataset(filename=filename, bucket_name=bucket_name)
-
-    logger.info("Dataset obtained successfully.")
-    logger.debug(f"Dataset: {dataset}")
-
-    dataset_id = str(uuid.uuid4())
-    database.set_dataset(dataset_id=dataset_id, filename=filename, dataset=dataset)
-
-    logger.info("Dataset uploaded successfully.")
+    """
+    Check if the filename ends with '.json'.
+    Process the contents if it is a valid '.json' file or else log the error.
+    """
+    if filename[-5:].lower() == ".json":
```

Comment:
I think we can extract this conditional so that we don't have nested if statements within it. Like this:

if filename[-5].lower() != '.json':
    logger.error(f"Invalid filetype received - {filename}")
    return

That will make the code below this more maintainable as there is less coupling between this if statement and the logic within

Reply:
@JackRimmerAND I've corrected the if logic to the above, hope it looks better now. Also, about extracting the last 5 characters, I just checked and see that if the string itself is less than 5 characters, it just extracts the string itself, so no errors here.